### DISCUSSION TITLE
Haproxy and https/http redirect problems

### DISCUSSION THREAD

**Discussion by gittbuh** [kimai/kimai#2792] [Questions & Answers]
Hi all,

I just installed kima2 in a lxc container inside proxmox running ubuntu 20.04, following to the letter this tutorial mentionned in the documentation here: https://www.kimai.org/documentation/fresh-ubuntu-20.html. Everything went very smoothly, no errors. I can access my kimai instance locally at a local ip address and also externally through haproxy (although this is problematic, as I explain below). 

I have used pfsense to create a letsencrypt certificate and haproxy (also in pfsense) to redirect port 80 to port 443/https to be able to access securely from the outside through a subdomain, as I usually do with many other services. However, I am having issues with redirects it seems. When I  try to access my kimai instance normally at https://subdomain.domain.com, it times out at subdomain.domain.com/en/login. I manage to see the login screen at https://subdomain.domain.com/en/login but when I click the login button, it times out again at subdomain.domain.com/homepage; at that point, if I add https before the url, it will login normally and everything seems to work ok, until I logout, then it times out again at subdomain.domain.com/homepage. 

Following the instructions in the "Reverse proxy" section of the documention here https://www.kimai.org/documentation/webserver-configuration.html), I copied  my kimai instace domain (subdomain.domain.com) into the trusted_proxies and trusted_hosts in the .env file, but this did not make any difference. 

Any tips on how to solve this situation would be really appreciated. 

Thanks!

**Comment by kevinpapst** [comment]
Doesn't sound like a Kimai problem. Did you set the required environment `X-Forwarded` variables, so Kimai can detect the original protocol as described in https://www.kimai.org/documentation/webserver-configuration.html#reverse-proxy?
What do the log files (webserver, php-fpm, kimai) say?

**Reply by gittbuh** [reply]
Thanks a lot for the reply. I had checked that instruction regarding reverse proxies before but I did not understand it very well unfortunately. Where do I set those variables exactly? In Kimai itself or in the reverse proxy? I use haproxy inside pfsense as reverse proxy, I do everything through the graphical ui but there is an advanced settings section I have never used where I can pass thru text to the backend for example.

As for logs, I looked in Doctor and checked the /var/log/. Maybe it is normal but cannot see the log in Doctor (I get a "missing log file" message) and there is no prod.log (this is the kimai log, right?) either under /var/log. I checked the php-fpm and found the follwing lines repeat over and over:

[27-Sep-2021 23:08:41] NOTICE: fpm is running, pid 108
[27-Sep-2021 23:08:41] NOTICE: ready to handle connections
[27-Sep-2021 23:08:41] NOTICE: systemd monitor interval set to 10000ms

As for nginx logs, in the access.log there is nothing special, and in the error.og I found these lines from yesterday:

2021/09/26 01:44:24 [notice] 30505#30505: signal process started
2021/09/26 01:47:04 [notice] 30518#30518: signal process started
2021/09/26 10:59:33 [warn] 107#107: server name "https://mysubdomain.mydomain.com" has suspicious sym
bols in /etc/nginx/sites-enabled/kimai2:4
2021/09/26 10:59:33 [warn] 119#119: server name "https://mysubdomain.mydomain.com" has suspicious sym
bols in /etc/nginx/sites-enabled/kimai2:4

Please let me know if there is any other info I can provide.

**Reply by kevinpapst** [reply]
No log file usually means "everything is good" if there is no warning in the permission section.

Said that: you are using a rather complex setup and should know how to configure it. If not, please go ahead and read the pfsense & haproxy docs how to set the variables.

**Reply by gittbuh** [reply]
ok thanks, I will do that. I take the opportunity to thank you for creating and making available for free kimai 2 :)
